services/src/main/java/org/fao/geonet/services/metadata/format/cache/FilesystemStore.java

```diff
@@ -0,0 +1,336 @@
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.PreDestroy;
+
+import static org.fao.geonet.constants.Params.Access.PRIVATE;
+import static org.fao.geonet.constants.Params.Access.PUBLIC;
+
+/**
+ * A {@link org.fao.geonet.services.metadata.format.cache.PersistentStore} that saves the files to disk.
+ *
+ * @author Jesse on 3/5/2015.
+ */
+public class FilesystemStore implements PersistentStore {
+    private static final String BASE_CACHE_DIR = "formatter-cache";
+    private static final String INFO_TABLE = "info";
+    private static final String KEY = "keyhash";
+    private static final String CHANGE_DATE = "changedate";
+    private static final String PUBLISHED = "published";
+    private static final String PATH = "path";
+    private static final String STATS_TABLE = "stats";
+    private static final String NAME = "name";
+    private static final String CURRENT_SIZE = "currentsize";
+    private static final String VALUE = "value";
+    public static final String WITHHELD_MD_DIRNAME = "withheld_md";
+    public static final String FULL_MD_NAME = "full_md";
+
+    private static final String QUERY_GET_INFO = "SELECT * FROM " + INFO_TABLE + " WHERE " + KEY + "=?";
+    private static final String QUERY_GET_INFO_FOR_RESIZE = "SELECT " +KEY + "," + PATH + " FROM " + INFO_TABLE + " ORDER BY " + CHANGE_DATE + " ASC";
+    private static final String QUERY_PUT = "MERGE INTO " + INFO_TABLE + " (" + KEY + "," + CHANGE_DATE + "," + PUBLISHED + "," + PATH + ") VALUES (?,?,?, ?)";
+    private static final String QUERY_REMOVE = "DELETE FROM " + INFO_TABLE + " WHERE " + KEY + "=?";
+    public static final String QUERY_SETCURRENT_SIZE = "MERGE INTO "+STATS_TABLE + " (" + NAME + ", " + VALUE + ") VALUES ('" + CURRENT_SIZE + "', ?)";
+    public static final String QUERY_GETCURRENT_SIZE = "SELECT "+VALUE+" FROM " + STATS_TABLE + " WHERE "+NAME+" = '"+CURRENT_SIZE+"'";
+
+    @Autowired
+    private GeonetworkDataDirectory geonetworkDataDir;
+    @VisibleForTesting
+    Connection metadataDb;
+    private boolean testing = false;
+    private volatile long maxSizeB = 10000;
+    private volatile long currentSize = 0;
+    private volatile boolean initialized = false;
+
+    private synchronized void init() throws SQLException {
+        if (!initialized) {
+            // using a h2 database and not normal geonetwork DB to ensure that the accesses are always on localhost and therefore
+            // hopefully quick.
+            try {
+                Class.forName("org.h2.Driver");
+            } catch (ClassNotFoundException e) {
+                throw new Error(e);
+            }
+
+            String[] initSql = {
+                    "CREATE SCHEMA IF NOT EXISTS " + INFO_TABLE,
+                    "CREATE TABLE IF NOT EXISTS " + INFO_TABLE + "(" + KEY + " INT PRIMARY KEY, " + CHANGE_DATE + " BIGINT NOT NULL, " +
+                    PUBLISHED + " BOOL NOT NULL, " + PATH + " VARCHAR(256) NOT NULL)",
```

Comment:
I wonder if 256 chars will be enough to describe the path (in some complex cases) ?